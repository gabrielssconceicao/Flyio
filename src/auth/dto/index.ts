import { LoginDto } from './login.dto';
import { LoginResponseDto } from './login-response.dto';
import { RefreshTokenDto } from './refresh-token.dto';
import { TokenPayloadDto } from './token-payload.dto';

export { LoginDto, LoginResponseDto, RefreshTokenDto, TokenPayloadDto };
