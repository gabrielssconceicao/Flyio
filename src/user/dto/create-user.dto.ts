import { ApiProperty } from '@nestjs/swagger';
import {
  IsEmail,
  IsNotEmpty,
  IsOptional,
  IsString,
  MaxLength,
  MinLength,
} from 'class-validator';
export class CreateUserDto {
  @ApiProperty({
    example: 'John Doe',
    description: 'User name',
    minLength: 3,
    maxLength: 255,
  })
  @IsString()
  @IsNotEmpty()
  @MinLength(3)
  @MaxLength(255)
  name: string;

  @ApiProperty({
    example: 'jdoe@email.com',
    description: 'User email',
  })
  @IsEmail()
  email: string;

  @ApiProperty({
    example: '123456',
    description: 'User password',
    minLength: 6,
    maxLength: 255,
  })
  @IsString()
  @IsNotEmpty()
  @MinLength(6)
  @MaxLength(255)
  password: string;

  @ApiProperty({
    example: 'This is my bio',
    description: 'User bio',
    maxLength: 255,
  })
  @IsOptional()
  @IsString()
  @MaxLength(255)
  bio?: string;

  @ApiProperty({
    example: 'example.png',
    description: 'User profile image',
  })
  @IsOptional()
  @IsString()
  profileImg?: string;
}
