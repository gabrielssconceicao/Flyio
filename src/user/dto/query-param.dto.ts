import { Type } from 'class-transformer';
import { IsOptional, IsString, Max, Min } from 'class-validator';

export class QueryParamDto {
  @IsOptional()
  @IsString()
  search?: string;
  @IsOptional()
  @Min(1)
  @Max(50)
  @Type(() => Number)
  limit?: number;
  @IsOptional()
  @Min(0)
  @Type(() => Number)
  offset?: number;
}
