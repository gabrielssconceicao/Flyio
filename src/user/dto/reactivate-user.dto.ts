import { ApiProperty } from '@nestjs/swagger';
import { IsNotEmpty, IsString } from 'class-validator';
import {
  dtoErrorMessages,
  DtoErrorType,
} from '../../common/utils/dto-error-messages';

export class ReactivateUserDto {
  @ApiProperty({
    example: 'krcio48hgGYIFYJfuyfJYAYu',
    description: 'Activation token',
  })
  @IsNotEmpty({ message: dtoErrorMessages('Token', DtoErrorType.REQUIRED) })
  @IsString()
  token: string;
}
