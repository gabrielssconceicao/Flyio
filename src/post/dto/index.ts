import { CreatePostDto } from './create-post.dto';
import { FindAllPostsResponseDto } from './find-all-posts.dto';
import { FindAllLikedPostsResponseDto } from './find-all-liked-post.dto';
export { CreatePostDto, FindAllPostsResponseDto, FindAllLikedPostsResponseDto };
