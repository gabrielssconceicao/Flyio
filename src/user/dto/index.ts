import { CreateUserDto } from './create-user.dto';
import { FindAllUsersResponseDto } from './find-all-users.dto';
import { QueryParamDto } from './query-param.dto';
import { ReactivateUserDto } from './reactivate-user.dto';
import { UpdateUserDto } from './update-user.dto';

export {
  CreateUserDto,
  FindAllUsersResponseDto,
  QueryParamDto,
  ReactivateUserDto,
  UpdateUserDto,
};
