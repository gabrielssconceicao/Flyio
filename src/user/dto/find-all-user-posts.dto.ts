import { ApiProperty, OmitType } from '@nestjs/swagger';
import { FindAllResponseDto } from 'src/common/dto/find-all-response.dto';
import { PostEntity } from '../../post/entities/post.entity';

export class FindAllUsersPostsResponseDto extends FindAllResponseDto<
  Omit<PostEntity, 'user'>
> {
  @ApiProperty({
    type: [OmitType(PostEntity, ['user'])],
    description: 'List of posts of a user with limited details',
  })
  items: Omit<PostEntity, 'user'>[];
}
