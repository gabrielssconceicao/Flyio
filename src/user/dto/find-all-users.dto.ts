import { ApiProperty } from '@nestjs/swagger';
import { PickType } from '@nestjs/mapped-types';
import { User } from '../entities/user.entity';
import { FindAllResponseDto } from '../../common/dto/find-all-response.dto';

class UserSummaryDto extends PickType(User, [
  'id',
  'name',
  'username',
  'profileImg',
] as const) {
  @ApiProperty({
    example: 'd-43-5df-df4',
    description: 'Unique identifier of the user',
  })
  id: string;

  @ApiProperty({ example: 'John Doe', description: 'Name of the user' })
  name: string;

  @ApiProperty({ example: 'jdoe', description: 'Username of the user' })
  username: string;

  @ApiProperty({
    example: 'https://example.com/profile.jpg',
    description: 'Profile image URL',
    nullable: true,
  })
  profileImg: string | null;
}

export class FindAllUsersResponseDto extends FindAllResponseDto<UserSummaryDto> {
  @ApiProperty({
    type: [UserSummaryDto],
    description: 'List of users with limited details',
  })
  items: UserSummaryDto[];
}
