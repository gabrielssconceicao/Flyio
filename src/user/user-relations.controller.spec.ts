import { Test, TestingModule } from '@nestjs/testing';
import { JwtService } from '@nestjs/jwt';
import { UserRelationsController } from './user-relations.controller';
import { UserRelationsService } from './user-relations.service';
import jwtConfig from '../auth/config/jwt.config';
import {
  jwtServiceMock,
  jwtConfigurationMock,
  generateTokenPayloadDtoMock,
} from '../auth/mocks';
import { PaginationDto } from '../common/dto/pagination.dto';
import { FindAllPostsResponseDto } from '../post/dto';
import { generateFindAllPostsDtoMock } from '../post/mock';

describe('UserRelationsController', () => {
  let controller: UserRelationsController;
  let service: UserRelationsService;

  let username: string;
  let paginationDto: PaginationDto;
  let findAllPosts: FindAllPostsResponseDto;
  beforeEach(async () => {
    const module: TestingModule = await Test.createTestingModule({
      controllers: [UserRelationsController],
      providers: [
        {
          provide: UserRelationsService,
          useValue: {
            reactivate: jest.fn(),
            getAllPostsByUsername: jest.fn(),
            getAllLikedPostsByUsername: jest.fn(),
          },
        },

        {
          provide: JwtService,
          useValue: jwtServiceMock,
        },
        {
          provide: jwtConfig.KEY,
          useValue: jwtConfigurationMock,
        },
      ],
    }).compile();

    controller = module.get<UserRelationsController>(UserRelationsController);
    service = module.get<UserRelationsService>(UserRelationsService);
    username = 'username';
    paginationDto = {
      offset: 1,
      limit: 10,
    };
    findAllPosts = generateFindAllPostsDtoMock();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should be defined', () => {
    expect(controller).toBeDefined();
    expect(service).toBeDefined();
  });

  describe('<Reactivate />', () => {
    it('should reactivate a user successfully', async () => {
      const reactivateUserDto = {
        token: 'token',
      };
      const message = 'User reactivated successfully';
      jest.spyOn(service, 'reactivate').mockResolvedValue({ message });
      const result = await controller.reactivate(reactivateUserDto);
      expect(service.reactivate).toHaveBeenCalledWith(reactivateUserDto);
      expect(result).toEqual({ message });
      expect(result).toMatchSnapshot();
    });
  });

  describe('<GetAllPostsByUsername />', () => {
    it('should get all posts by username successfully', async () => {
      jest
        .spyOn(service, 'getAllPostsByUsername')
        .mockResolvedValue(findAllPosts);
      const result = await controller.getAllPostsByUsername(
        username,
        paginationDto,
        generateTokenPayloadDtoMock(),
      );
      expect(service.getAllPostsByUsername).toHaveBeenCalledWith(
        username,
        paginationDto,
        generateTokenPayloadDtoMock(),
      );
      expect(result).toMatchSnapshot();
    });
  });

  describe('<GetAllLikedPostsByUsername />', () => {
    it('should get all posts by username successfully', async () => {
      findAllPosts.items = findAllPosts.items.map((post) => ({
        ...post,
        liked: true,
      }));
      jest
        .spyOn(service, 'getAllLikedPostsByUsername')
        .mockResolvedValue(findAllPosts);
      const result = await controller.getAllPostsByUsername(
        username,
        paginationDto,
        generateTokenPayloadDtoMock(),
      );
      expect(service.getAllPostsByUsername).toHaveBeenCalledWith(
        username,
        paginationDto,
        generateTokenPayloadDtoMock(),
      );
      expect(result).toMatchSnapshot();
    });
  });
});
