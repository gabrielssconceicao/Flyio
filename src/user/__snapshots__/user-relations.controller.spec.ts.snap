// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserRelationsController <GetAllLikedPostsByUsername /> should get all posts by username successfully 1`] = `undefined`;

exports[`UserRelationsController <GetAllPostsByUsername /> should get all posts by username successfully 1`] = `
{
  "count": 1,
  "items": [
    {
      "content": "Hello world",
      "createdAt": 2023-03-15T17:30:00.000Z,
      "id": "42-d-f-df4",
      "images": [],
      "liked": false,
      "likes": 0,
      "user": {
        "profileImg": "https://example.com/profile.jpg",
        "username": "jDoe453",
      },
    },
  ],
}
`;

exports[`UserRelationsController <Reactivate /> should reactivate a user successfully 1`] = `
{
  "message": "User reactivated successfully",
}
`;
